### PR TITLE
Added non-agent-specific docs on span compression

### PR DESCRIPTION
**Pull request by beniwohli**
## Motivation/summary

The agent team felt that the [span compression feature](https://github.com/elastic/apm/blob/master/specs/agents/handling-huge-traces/tracing-spans-compress.md) requires some documentation that is not agent specific. Similar to the [sampling docs](https://github.com/elastic/apm-server/blob/master/docs/sampling.asciidoc), it seems like the APM Server docs is the best place for this.

## Checklist

- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)
- [x] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## Related issues

https://github.com/elastic/apm/issues/432
#5485

### CODE REVIEW

**Comment by simitt** [comment]
@beniwohli can we help to get this PR out of Draft state or is it blocked by something?

**Comment by beniwohli** [comment]
@simitt totally forgot that it is still in draft, thanks for the heads up. I'll give it a last read through this morning and set it to ready for review afterwards 👍

**Comment by beniwohli** [comment]
The references to the Go agent docs break because the feature is not released yet and hence only available in the `master` docs, not `current`. I'm not sure how to handle this properly.

**Comment by simitt** [comment]
@bmorelli25 could you help out on this issue please. 
> The references to the Go agent docs break because the feature is not released yet and hence only available in the master docs, not current. I'm not sure how to handle this properly.

File: docs/span-compression.asciidoc [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+Spans with longer duration are not compressed. Please refer to the agent documentation for specifics.
+
+[float]
+[[span-compression-support]]
+=== Agent support
+
+Support for span compression is available in these agents:
+
+ * Go: {apm-go-ref}/configuration.html#config-span-compression-exact-match-duration[`ELASTIC_APM_SPAN_COMPRESSION_EXACT_MATCH_MAX_DURATION`], {apm-go-ref}/configuration.html#config-span-compression-same-kind-duration[`ELASTIC_APM_SPAN_COMPRESSION_SAME_KIND_MAX_DURATION`]
+ * Python: {apm-py-ref}/configuration.html#config-span-compression-exact-match-max_duration[`span_compression_exact_match_max_duration`], {apm-py-ref}/configuration.html#config-span-compression-same-kind-max-duration[`span_compression_same_kind_max_duration`]
```
Comment (bmorelli25) [inline docs/span-compression.asciidoc:65]: In these cases, I either comment out the links for now, or I hard-code the links to the master branch and open an issue to change them when the feature releases.

```suggestion
 * Go: https://www.elastic.co/guide/en/apm/agent/go/master/configuration.html#config-span-compression-exact-match-duration[`ELASTIC_APM_SPAN_COMPRESSION_EXACT_MATCH_MAX_DURATION`], https://www.elastic.co/guide/en/apm/agent/go/master/configuration.html#config-span-compression-same-kind-duration[`ELASTIC_APM_SPAN_COMPRESSION_SAME_KIND_MAX_DURATION`]
 * Python: {apm-py-ref}/configuration.html#config-span-compression-exact-match-max_duration[`span_compression_exact_match_max_duration`], {apm-py-ref}/configuration.html#config-span-compression-same-kind-max-duration[`span_compression_same_kind_max_duration`]
```

**Comment by bmorelli25** [comment]
Pushed the fix and opened https://github.com/elastic/observability-docs/issues/1285 to update my workaround when the next version of the Go Agent releases.

**Review by bmorelli25 — APPROVED** [review]

**Pull request merged**